Remotion/Core/Collections.Caching/CacheFactory.cs

```diff
@@ -276,7 +289,9 @@ public static class CacheFactory
     /// should therefor be replaced.
     /// </remarks>
     [Obsolete ("Use CreateWithSynchronization(...) instead. (Version: 1.19.3)")]
-    public static ICache<TKey, TValue> CreateWithLazyLocking<TKey, TValue> () where TValue : class
+    public static ICache<TKey, TValue> CreateWithLazyLocking<TKey, TValue> ()
+        where TKey : notnull
+        where TValue : class

```

Comment:
https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/generics/constraints-on-type-parameters

I just found the class? constraint, allowing for nullable OR not-nullable reference types. Wouldn't this be the more correct version to use for most places where we use reference types as generic parameters and null is not forbidden as a value, i.e. there is no argument check. And yes, this means we have to recheck the other branches, too :/